### ISSUE TITLE
os-tor: allow ControlPort 0.0.0.0:9051

### ISSUE THREAD

**Issue by Sjors** [opnsense/plugins#3412]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] When the request is meant for an existing plugin, I've added its name to the title.

**Is your feature request related to a problem? Please describe.**

I'd like to move a (not really security critical) Tor hidden service from a server to the router. The service itself still runs on the server, but I'd like it to use Tor on the router.

**Describe the solution you'd like**

Allow 0.0.0.0:9051 for the ControlPort. This is allowed (albeit discouraged) in torrc. The GUI does not allow entering this value, because I think validation constrains it to a port number. 

**Describe alternatives you've considered**
Manually change it in torrc. This works, but I assume it's lost every reboot? Manually update in config.xml. This probably works (haven't tried a restart) because it's not revalidated as long as you don't touch the field in the GUI.

A more tightly controlled list of interfaces, like for the SOCKS proxy, would be better of course. But afaik the control port can only bind to one interface.

**Additional context**
<img width="604" alt="Scherm­afbeelding 2023-04-29 om 23 25 49" src="https://user-images.githubusercontent.com/10217/235324916-ea447905-b15c-47aa-861d-368e9a3b5506.png">

https://stackoverflow.com/a/58138489/313633

OPNsense version 23.1.6-amd64
os-tor 1.9

**Comment by fabianfrz** [comment]
> I'd like to move a (not really security critical) Tor hidden service from a server to the router. The service itself still runs on the server, but I'd like it to use Tor on the router.

The plugin supports hidden services via the GUI. No need for opening the control port.


Alternatives:

Do what it suggests - connect via SSH to OPNsense and use a port forward. This port is not intended to be opened to the outside world. BTW: The plugin uses this port internally for querying data. So if you change something there, you may break the diagnostics pages. A NAT rule or a load balancer (nginx, haproxy in tcp mode) would work as well but you should not do that.

**Comment by Sjors** [comment]
I indeed noticed the diagnostic page broke with my approach.

I have one application that spins up its own hidden service using the control port. But it's not too bad to instead let the router create it and just point the application to it.

To move an existing hidden service, I ended up creating one on the device and then replacing its hostname, privkey and pubkey with the desired one. It would be nice to have an import feature for that, but it's doable.
